### DISCUSSION TITLE
SmilesMolSupplier parameter

### DISCUSSION THREAD

**Discussion by gwbgithub** [rdkit/rdkit#4392] [Ideas] [answered]
I'm using a csv file as input, and would like a parameter like quotechar from the pandas read_csv function.  Right now my csv has a chemical name field that sometimes contains commas (and is quoted) and the commas in the chemical names are being read as field separators.  I would like "blah,1,2-blah" to be read in as a single field, even when my delimiter is a comma.

**Comment by bp-kelley** [comment, marked as answer]
SmilesMolSupplier doesn't properly quote the delimiters (or quotes) inside quotes.

Until this is fixed, which it very well might not be, you will probably have better luck 
(1) loading the csv with pandas and then getting the data out

or
(2) using python's csv module which does appropriately handle commas and quotes.

 I mostly do something like the following.

```
import csv
import logging
from rdkit import Chem
def read_csv_with_mols(filename, smiles_col="smiles"):
    with open(filename) as f:
        reader = csv.reader(f)
        for lineno, data in enumerate(reader):
            if lineno == 0: # assume smiles holds the smiles column, might need to change
                smilesidx = data.index(smiles_col)
                header = data
            else:
                smiles = data[smilesidx]
                m = Chem.MolFromSmiles(smiles)
                if m:
                    for name, value in zip(header, data):
                        m.SetProp(name, value)
                    yield m
               else:
                  logging.warning("Could not parse smiles %s at line number %d, ignoring,...", smiles, lineno)
```
            

The alternative is to change your csv delimiter to tabs "\t" when you write it, then you can set the delimiter during read:

```
for mol in Chem.SmilesMolSupplier("foo.txi", delimiter='\t'):
```

Good luck.
